accepted/future-releases/extension-types/feature-specification.md

```diff
@@ -1377,21 +1365,20 @@ used as an expression *(also known as the ultimate representation type)*.
 ### Summary of Typing Relationships
 
 *Here is an overview of the subtype relationships of an extension type `V0`
-with instantiated representation type `R` and instantiated superinterface
-types `V1 .. Vk`, as well as other typing relationships involving `V0`:*
+with instantiated representation type `R` (whose extension type erasure is `R0`)
+and instantiated superinterface types `V1 .. Vk`, as well as other typing
+relationships involving `V0`:*
 
 - *`V0` is a proper subtype of `Object?`.*
-- *`V0` is a supertype of `Never`.*
-- *If `R` is a non-nullable type then `V0` is a proper subtype of
-  `Object`, and a non-nullable type.*
+- *`V0` is a proper supertype of `Never`.*
 - *`V0` is a proper subtype of each of `V1 .. Vk`.*
-- *At run time, the type `V0` is identical to the type `R`. In
-  particular, `o is V0` and `o as V0` have the same dynamic
-  semantics as `o is R` respectively `o as R`, and
+- *At run time, the type `V0` is identical to the extension type erasure
+  `R0`. In particular, `o is V0` and `o as V0` have the same dynamic
+  semantics as `o is R0` respectively `o as R0`, and
   `t1 == t2` evaluates to true if `t1` is a `Type` that reifies
-  `V0` and `t2` reifies `R`, and the equality also holds if
-  `t1` and `t2` reify types where `V0` and `R` occur as subterms
-  (e.g., `List<V0>` is equal to `List<R>`).*
+  `V0` and `t2` reifies `R0`, and the equality also holds if
```

Comment:
I changed several things in this item, at was somewhat clunky. However, I left one thing unchanged:

> "`Type` that reifies `V0`" is not a thing

An extension type `E` has a run-time representation and semantics (which is the same as that of the extension type erasure of `E`), which in turn corresponds to a specific set of instances of `Type` (a single canonical instance in some cases).

I don't actually think it's going to help anybody if we insist that we should say "let `t1` be a `Type` that reifies the extension type erasure of `V0`" rather than "let `t1` be a `Type` that reifies `V0`".

Developers will see `return T;` or similar constructs where we usually talk about a `Type` that reifies the type `T`, and this relationship doesn't get destroyed by the fact that we take two steps to get from an extension type to the corresponding `Type`, rather than one. In any case, `V0` is a type in a Dart program, and there is a corresponding set of `Type` instances at run time, and we usually use the word 'reify' to characterize that relationship.

Reply:
We needs some kind of verbage for this, probably in order to write the runtime semantics too.

If we have a type literal expression *e*, the runtime semantics of evaluating it should be something like:
* Let *T* be the static type denoted by the type expression *e*. (This includes type inference of missing type variables).
* Let *R* be the runtime type corresponding to *T*. *R* is created from *T* by:
    * Perform extension type erasure on *T* to get *T'*. 
    * If *T'* contains type variables, those are replaced by the runtime type currently bound to that type variable.
    * The result of that is *R*.
* Let *v* be a constant `Type` object reifying *R*.

The "runtime type corresponding to *T*" is needed for type variables already. Don't know how we've phrased it.

Reply:
> The "runtime type corresponding to T" is needed for type variables already. Don't know how we've phrased it.

Right, we do have that part, it's described as 'the actual type' [here](https://github.com/dart-lang/language/blob/8a4003d2d773e324d1a2ac7c9bbb553260f5cd5b/specification/dartLangSpec.tex#L22769).

Reply:
Note that the 'Summary of Typing Relationships' is commentary, there is no normative text in that section.

Reply:
I simplified this considerably, and it doesn't talk about reification any more.